File: LeetCode/1784/golang/lc1784_test.go
package lc1784

import "testing"

func Test_checkOnesSegment(t *testing.T) {
	testcases := []struct {
		s        string
		expected bool
	}{
		{"1001", false},
		{"10101", false},
		{"110", true},
		{"1", true},
		{"10000000", true},
		{"11111110", true},
		{"11111111", true},
	}

	for i, tc := range testcases {
		actual := checkOnesSegment(tc.s)
		if actual != tc.expected {
			t.Errorf("Testcase checkOnesSegment#%02d (%s) failed: want %t, got %t",
				i+1, tc.s, tc.expected, actual)
		}
	}
}

File: LeetCode/1784/python/lc1784_test.py
import unittest

from lc1784 import Solution

class Test_checkOnesSegment(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertEqual(s.checkOnesSegment("1001"), False)
        self.assertEqual(s.checkOnesSegment("10101"), False)
        self.assertEqual(s.checkOnesSegment("110"), True)
        self.assertEqual(s.checkOnesSegment("1"), True)
        self.assertEqual(s.checkOnesSegment("11111111111"), True)
        self.assertEqual(s.checkOnesSegment("10000000000"), True)
        self.assertEqual(s.checkOnesSegment("11111111110"), True)

if __name__ == "__main__":
    unittest.main()
